### ISSUE TITLE
[Feature]: Add multiple gyms to validation and testing

### ISSUE THREAD

**Issue by alfieroddanintel** [open-edge-platform/physical-ai-studio#136]
### Problem Statement

Our dataloader init currently looks like this:

```python
class DataModule(LightningDataModule):
    def __init__(...)
        # dataset
        self.train_dataset: Dataset = train_dataset
        self.train_batch_size: int = train_batch_size

        # gym environments
        self.val_gym: Gym | None = val_gym
        self.val_dataset: Dataset | None = None
        self.num_rollouts_val: int = num_rollouts_val
        self.test_gym: Gym | None = test_gym
        self.test_dataset: Dataset | None = None
        self.num_rollouts_test: int = num_rollouts_test
        self.max_episode_steps = max_episode_steps
```
And we would train a model with something like this:

```python
if __name__ == "__main__":
    gym = PushTGym()
    datamodule = LeRobotDataModule(
        repo_id="lerobot/pusht",
        train_batch_size=128,
        val_gym=gym,
        test_gym=gym,
        num_rollouts_val=10,
        num_rollouts_test=30,
    )
    logger = TensorBoardLogger(save_dir="experiments")
    trainer = Trainer(max_epochs=300, logger=logger, check_val_every_n_epoch=10)
    policy = ACT()
```

This is fine for a single dataset to gym combination.

What would be great is when we have multiple task suites like in LIBERO:

```python
task_suites=["libero_spatial", "libero_object", "libero_goal"]
```

It would be an advantage to supply each gym individually to validation and testing.

```
val_gyms = [LiberoGym(task_suite="task") for task in task_suites]
datamodule = LeRobotDataModule(
        repo_id="libero_multitask",
        train_batch_size=128,
        val_gyms=val_gyms,
        num_rollouts_val=10,
    )
```

### Proposed Solution

```python
def val_dataloader(self) -> DataLoader[Any]:
    return [
                DataLoader(
                    ds,
                    batch_size=1,
                    collate_fn=_collate_gym,
                    shuffle=False,
                )
                for ds in self.test_datasets
            ]

```
We make use of [pytorch lightning list of dataloaders feature](https://lightning.ai/docs/pytorch/LTS/guides/data.html#multiple-validation-test-predict-dataloaders), then we can return a dataloader per task like:

```bash
──────────────────────────────────────────────────────────────────────────────────
        Test metric               DataLoader 0               DataLoader 1
──────────────────────────────────────────────────────────────────────────────────
test/gym/avg_episode_length           33.25                      33.25
  test/gym/avg_max_reward              1.0                        1.0
  test/gym/avg_sum_reward       32.24305725097656          32.24305725097656
```

(we can add naming dataloaders)

In order to do this we must change the signature of `validation_step` and `test_step`:

```python
def validation_step(self, batch: Gym, batch_idx: int, dataloader_idx=0)

def test_step(self, batch: Gym, batch_idx: int, dataloader_idx=0) 
```

### Component

Library (VLA framework)

### Alternatives Considered

- We could make use of `ConcatDataset` from torch, this would require each `GymDataset` returning a name of the gym and the metric using this information accordingly.

### Acceptance Criteria

- [ ] We are able to report multiple metrics on multiple gyms in validation / test and separate out the metrics per gym. This should be done in one `.fit` or `.test` call.
